espfs: compatibility with `documentation` change

xobs opened the pull request:
The API of espfs was changed with the commit numbered
[39ca280970e47b4dcc6081db460a3990e7b537aa ](https://github.com/jkent/libespfs/commit/39ca280970e47b4dcc6081db460a3990e7b537aa)(accompanied solely by the
message `documentation`). This change was pushed on 29 December 2020, so it's been around almost a full year.

This commit renames symbols to match the new names so that `libesphttpd` works with the current version of `libespfs`, as well as every release since `v1.0.0`.

xobs committed:
espfs: compatibility with `documentation` change

The API of espfs was changed with the commit numbered
39ca280970e47b4dcc6081db460a3990e7b537aa (accompanied solely by the
message `documentation`).

This commit renames symbols to match the new names.

Signed-off-by: Sean Cross <sean@xobs.io>

phatpaul commented:
thanks xobs, this was also introduced in PR #99 

Unfortunately it breaks compatibility with older espfs used by https://github.com/chmorgan/esphttpd-freertos (and my project).

I'm stuck on the old espfs because I haven't been able to get the later versions to build on Make (MSYS2 on windows).  Actually @jkent removed support for Make in libespfs.  And I haven't been able to get [that project](https://github.com/chmorgan/esphttpd-freertos) to build with CMake.

If you can update [that project](https://github.com/chmorgan/esphttpd-freertos) to work with CMake, then I'll be able to test this patch.

phatpaul commented on core/httpd-espfs.c:
```diff
@@ -135,20 +134,20 @@ serveStaticFile(HttpdConnData *connData, const char* filepath) {
 
 		// Check if requested file was GZIP compressed
-		isGzip = espFsFlags(file) & FLAG_GZIP;
+		isGzip = espfs_ftell(file) & ESPFS_FLAG_GZIP;
```
I don't think ftell is the right function to use here.  Documentation for ftell is to return the file position, not the status bits.
M-bab has a different solution using fstat: https://github.com/chmorgan/libesphttpd/pull/99/files#diff-6d88c08ed161a2002c3e087b42d85902444d46ed1077cb3420356da699753cbbR142

phatpaul commented on core/httpd-espfs.c:
```diff
@@ -353,9 +352,9 @@ CgiStatus ICACHE_FLASH_ATTR cgiEspFsTemplate(HttpdConnData *connData) {
 		tpd->tplArg=NULL;
 		tpd->tokenPos=-1;
-		if (espFsFlags(tpd->file) & FLAG_GZIP) {
+		if (espfs_ftell(tpd->file) & ESPFS_FLAG_GZIP) {
```
same issue as above

xobs commented:
You're right, I ended up pulling that into its own code in https://github.com/xobs/libesphttpd/commit/4665561f1d3b9e9a4f79c0b6272e04bedc776933, after running `clang-format` on the repository.

I've since switched to using the built-in http server, though, so I won't do much else with libesphttpd.

phatpaul commented:
@xobs I've also been interested in the built-in https in esp-idf, but it seems to be a lower-level library.  Have you found any example or component that provides the higher level functions like this libesphttpd does, but internally uses the esp-idf http_server?  I.e. integrates the static file serving (using espfs or equiv.), OTA upgrade api, WiFi api, filesystem, etc.

xobs commented:
The only feature I don't yet have is the OTA upgrade API, but I'll work on that at some point because I feel it's important. I have the other features in your list.

Wifi API: I just delegate these to `esp-wifi-manager`. It used to integrate the web server internally, but I stripped that out to use libesphttpd. The interface between the two was simple enough to patch over, and when I moved to esp-idf 5.0 and switched back to http_server it was very easy to adapt. Unfortunately the upstream version appears abandoned, so I've forked it and am continuing development on my branch: https://github.com/farpatch/esp32-wifi-manager

Filesystem: I have a function that integrates the basics of frogfs, adapted from libesphttpd that works well. The biggest problem is the lack of progressive content, which I need to fake with chunked transfers. In this regard, libesphttpd is clearly superior, particularly when streaming infinite-length files. However, this solution works for my usecase: https://github.com/farpatch/farpatch/blob/main/main/http.c#L342-L400

Websockets: This is the biggest issue I've had, because libesphttpd does a lot behind the scenes to make websockets work well, particularly with broadcasting data. I've had to fake it a lot, and I'm still not entirely happy with the results. It works well enough to stream serial data, though: https://github.com/farpatch/farpatch/blob/main/main/websocket.c
